views/building-hours/building.js

```diff
@@ -16,75 +15,106 @@ import * as c from '../components/colors'
 import type {BuildingInfoType} from './types'
 import {isBuildingOpen} from './is-building-open'
 import {formatBuildingHours} from './format-building-hours'
+import CollapsibleBlock from '../components/collapsibleBlock'
+import {allBuildingHours} from './all-building-hours.js'
+import Dimensions from 'Dimensions'
 
-let styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    borderLeftWidth: 5,
-    borderRightWidth: 5,
-    margin: 5,
-    height: 100,
-  },
-  buildingOpen: {
-    borderColor: c.pastelGreen,
-  },
-  buildingClosed: {
-    borderColor: c.strawberry,
-  },
-  buildingAlmostClosed: {
-    borderColor: c.mustard,
-  },
-  name: {
-    color: c.white,
-    fontSize: 24,
-    textAlign: 'center',
-    backgroundColor: 'rgba(0, 0, 0, 0.65)',
-  },
-  inner: {
-    flex: 1,
-    flexDirection: 'column',
-    justifyContent: 'space-between',
-  },
-  status: {
-    color: c.white,
-    fontSize: 14,
-    paddingTop: 2,
-    paddingBottom: 2,
-    textAlign: 'center',
-    backgroundColor: 'rgba(0, 0, 0, 0.45)',
-  },
-})
+
+let openStatus
+let borderColor
 
 let borderColors = {
-  'Open': styles.buildingOpen,
-  'Almost Closed': styles.buildingAlmostClosed,
-  'Closed': styles.buildingClosed,
+  'Open': c.pastelGreen,
+  'Almost Closed': c.mustard,
+  'Closed': c.strawberry,
 }
 
 type PropsType = {
   info: BuildingInfoType,
   image: number,
   name: string,
 };
+
+
+
+let style = function(){
+  return {
+    container: {
+      flex: 1,
+      justifyContent: 'center',
+      borderLeftWidth: 5,
+      borderRightWidth: 5,
+      marginTop: 10,
+      height: 100,
+      borderColor: getBorderColor(),
+    },
+    name: {
+      color: c.white,
+      fontSize: 24,
+      textAlign: 'center',
+      backgroundColor: 'rgba(0, 0, 0, 0.65)',
+    },
+    inner: {
+      flex: 1,
+      flexDirection: 'column',
+      justifyContent: 'space-between',
+    },
+    status: {
+      color: c.white,
+      fontSize: 14,
+      paddingTop: 2,
+      paddingBottom: 2,
+      textAlign: 'center',
+      backgroundColor: 'rgba(0, 0, 0, 0.45)',
+    },
+    hoursText: {
+      textAlign: 'center',
+      fontSize: 13,
+    },
+    contents: {
+      paddingBottom: 10,
+      paddingTop: 5,
+      width: getContainerWidth(),
+    },
+  }
+}
+
+let getBorderColor = function(){
+  borderColor = borderColors[openStatus]
```

Comment:
I don't like global variables. Could you move this discovery process back into the BuildingView function, please?